kibana-reports/server/model/index.ts

```diff
@@ -113,7 +113,8 @@ export const reportDefinitionSchema = schema.object({
       schema.literal(REPORT_TYPE.savedSearch),
     ]),
     description: schema.string(),
-
+    header: schema.string(),
+    footer: schema.string(),
```

Comment:
If you add the fields here, which means header and footer are for all types of report. I don't think we need header/footer for csv report, can you confirm? If so, add header/footer fields inside the  `visualReportSchema`. If these fields are optional, use `header: schema.maybe(schema.string())`